### PR TITLE
[Typing][C-132,C-133,C-134] Add type annotations for `python/paddle/nn/{clip,decode,initializer/lazy_init}.py`

### PR DESCRIPTION
**Pull request by megemini**
### PR Category

User Experience

### PR Types

Improvements

### Description

类型标注：

- `python/paddle/nn/clip.py`
- `python/paddle/nn/decode.py`
- `python/paddle/nn/initializer/lazy_init.py`

### Related links
 
- https://github.com/PaddlePaddle/Paddle/issues/65008

@SigureMo @megemini

### CODE REVIEW

File: python/paddle/nn/initializer/lazy_init.py [hunk 1]
```diff
@@ -129,5 +133,5 @@ def __enter__(self):
         """
         lazy_init_helper().enable()
 
-    def __exit__(self, *args, **kwargs):
+    def __exit__(self, *args: object, **kwargs: Any) -> None:
```
Comment (SigureMo) [inline python/paddle/nn/initializer/lazy_init.py]: 这俩为啥一个 `object` 一个 `Any`

这里可以改成 `__exit__` 的标准签名么 https://stackoverflow.com/questions/58808055/what-are-the-python-builtin-exit-argument-types

Comment (megemini) [reply python/paddle/nn/initializer/lazy_init.py]: > 这俩为啥一个 `object` 一个 `Any`
> 

我也想知道啊 ... ...

``` shell
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1
- files were modified by this hook

Found 1 error (1 fixed, 0 remaining).

```

> 这里可以改成 `__exit__` 的标准签名么 https://stackoverflow.com/questions/58808055/what-are-the-python-builtin-exit-argument-types

我改一下吧，看看有木有问题 ～

Comment (SigureMo) [reply python/paddle/nn/initializer/lazy_init.py]: image

https://docs.astral.sh/ruff/rules/bad-exit-annotation/

好像是个 undocumented behavior，按照标准签名来吧，这个肯定没问题

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**